kore/src/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -337,7 +339,8 @@ andEqualsFunctions
         )
     => [(SimplificationTarget, TermTransformation variable unifier)]
 andEqualsFunctions = fmap mapEqualsFunctions
-    [ (AndT,    \_ _ _ _ -> boolAnd, "boolAnd")
+    [ (AndT,    \_ _ m s -> expandAlias (maybeTermAnd m s), "expandAlias")
```

Comment:
Why doesn't `expandAlias` use its `s` argument directly? Why is it hard-coded to `maybeTermAnd`?

Reply:
It's because `AndTerm` imports this file, so this file can't also import `AndTerm` to call `maybeTermAnd` itself, so we partially apply it and pass it down.